src/rust/runtime/network/socket/state.rs

```diff
@@ -97,7 +89,7 @@ impl SocketStateMachine {
             match self.may_connect() {
                 Ok(()) => {
                     // If either a time out or the current state changed, check it again.
-                    _ = self.current.clone().wait_for_change(Some(TIMEOUT)).await;
```

Comment:
The above comment doesn't hold true now. Same for other places where this comment is present.